### PR TITLE
updated nbt references | nbt preloading at start | Updated getItemInHand code

### PR DESCRIPTION
**Pull request by ricsal1**
NBT updated from 1.21.1
NBT preLoading at start (reduzes first shoot hang)
Updated player.getInventory().getItemInHand() to player.getInventory().getItemInMainHand()

### CODE REVIEW

**Review by Lorenzo0111 — CHANGES_REQUESTED** [review]
This will throw errors in legacy versions. Please rollback that

File: src/main/java/me/zombie_striker/qg/guns/Gun.java [hunk 1]
```diff
@@ -160,7 +160,7 @@ public static boolean USE_THIS_INSTEAD_OF_INDEVIDUAL_SHOOT_METHODS(Gun g, Player
     }
 
     public static int getAmount(Player player) {
-        return getAmount(player.getInventory().getItemInHand());
+        return getAmount(player.getInventory().getItemInMainHand());
```
Comment (Lorenzo0111) [inline src/main/java/me/zombie_striker/qg/guns/Gun.java:163]: This will throw errors in legacy versions. Please rollback that

File: pom.xml [hunk 2]
```diff
@@ -335,17 +342,11 @@
...
                 <version>3.2.2</version>
             </plugin>
-            <plugin>
-                <artifactId>maven-deploy-plugin</artifactId>
-                <version>3.1.1</version>
-                <configuration>
-                    <!--altDeploymentRepository>internal.repo::default::file://${project.build.directory}/mvn-repo</altDeploymentRepository-->
-                </configuration>
-            </plugin>
+
```
Comment (Lorenzo0111) [inline pom.xml:345]: Please put the old version, I need it for the CI and it already works fine

**Review by Lorenzo0111 — CHANGES_REQUESTED** [review]

File: src/main/java/me/zombie_striker/qg/guns/Gun.java (hunk 1 above)
Comment (ricsal1) [reply src/main/java/me/zombie_striker/qg/guns/Gun.java:163]: It exists on other classes, Ok with that?

File: pom.xml (hunk 2 above)
Comment (ricsal1) [reply pom.xml]: You meant this section?

**Pull request merged**